src/rdb.c

```diff
@@ -2306,19 +2306,33 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, redisDb* db, int rdbflags,
             * Expired hash fields on the master are silently discarded.
             * Note that if all fields in a hash has expired, the hash would not be
             * created in memory (because it is created on the first valid field), and
-            * thus the key would be discarded as an "empty key" */
-            if (expire != 0 && iAmMaster() && ((mstime_t)expire < now) &&  /* note: expire was saved to RDB as unix-time in milliseconds */
+            * thus the key would be discarded as an "expired hash" */
+            if (expireAt != 0 && iAmMaster() && ((mstime_t)expireAt < now) &&
                 !(rdbflags & RDBFLAGS_AOF_PREAMBLE))
             {
-                /* TODO: consider replication (like in rdbLoadAddKeyToDb) */
+                if (rdbflags & RDBFLAGS_FEED_REPL && db) {
```

Comment:
Perhaps this can be somehow optimized to include multiple expired fields in one key on a single `repl backlog` entry? Obviously not for now, but maybe open a ticket for that (or whatever)

Reply:
You mean, if all the fields got expired then we can discard the entire hash in a single command? If that so, then we need to iterate to the end of the hash and only then to decide whether to re-iterate and send multiple commands or a single command?

Reply:
That is one option, but perhaps `hdel` can support deleting multiple keys, even if not the entire hash?
But even so, aybe you can add `hdel` commands on the go and if you find out all the fields were expired, you can remove the hdel commands and add a `del` command for the key?

Reply:
Ok. I will mark it as TODO.